packages/time-picker/src/TimePicker/utils/getAllTimeOptions.ts

```diff
@@ -0,0 +1,26 @@
+import { Time } from "@internationalized/date"
+import { IncrementValues, TimeOption } from "../types"
+import { formatToLocalisedTime } from "./formatToLocalisedTime"
+
+type GetAllTimeOptionsConfig = {
+  locale: string
+  increments?: IncrementValues
+}
+
+export const getAllTimeOptions = ({
+  locale,
+  increments = 30,
+}: GetAllTimeOptionsConfig): Record<string, TimeOption> =>
+  Array.from(Array(24).keys()).reduce((options, hour) => {
```

Comment:
It doesn’t really matter, but it does feel a bit strange to use this "create an array of length N and throw it away" when this looks like a good use case for a `for (let hour = 0; hour <= 24; hour++)` style loop?